### PR TITLE
fix: mdapi deploy negative one

### PR DESCRIPTION
**Pull request by mshanemc**
### What does this PR do?
mdapi:deploy/retrieve/report has special treatment for `-w -1`
indefinitely long waits should be the default for retrieve/report

help text was reporting default twice

explanation:
This is in the plugin instead of SDR to avoid changing the behavior there and reduce the scope of this "feature" to just `mdapi:deploy`
this uses milliseconds because polling client *eventually* will be using ms.  using max_safe_int for something larger (ex: minutes) would probably break something when it gets converted to ms.

fix logger's fail on circular dependency https://github.com/forcedotcom/cli/issues/1518

### What issues does this PR fix or reference?
https://github.com/forcedotcom/cli/issues/1519
https://github.com/forcedotcom/cli/issues/1518
https://github.com/forcedotcom/cli/issues/1511 (about help being wrong)
[@W-11147622@](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE00000x9CzeYAE/view)
[@W-11150198@](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE00000x9rjzYAA/view)

### CODE REVIEW

**Comment by iowillhoit** [comment]
QA Notes
I was able to pass `-w -1` on all of the modified commands:
```
sfdx force:mdapi:deploy -d metadataPackage_1652718790193 -w -1
sfdx force:mdapi:deploy:report -w -1
sfdx force:mdapi:retrieve -r temp/retrieve-dir -w -1
```

**Review by iowillhoit — APPROVED** [review]

File: src/commands/force/mdapi/deploy/report.ts [hunk 1]
```diff
@@ -68,7 +71,7 @@ export class Report extends DeployCommand {
         : new DeployProgressStatusFormatter(this.logger, this.ux);
       progressFormatter.progress(deploy);
     }
-    this.deployResult = await deploy.pollStatus(500, this.getFlag<Duration>('wait').seconds);
```
Comment (iowillhoit) [inline src/commands/force/mdapi/deploy/report.ts:71]: Do you want to use the polling options object here too?

File: src/commands/force/mdapi/deploy/report.ts [hunk 2]
```diff
@@ -68,7 +71,7 @@ export class Report extends DeployCommand {
...
+    this.deployResult = await deploy.pollStatus(500, waitDuration.seconds);
```
Comment (iowillhoit) [inline src/commands/force/mdapi/deploy/report.ts:74]: Do you want to use the polling options object here too?

Comment (mshanemc) [reply src/commands/force/mdapi/deploy/report.ts:74]: yeah, lemme fix that, too

**Comment by iowillhoit** [comment]
Tested again with recent changes, LGTM

**Review by shetzel — APPROVED** [review]

**Comment by shetzel** [comment]
SOAP and REST
✅ `sfdx force:mdapi:deploy -d converted/ -l RunLocalTests -w -1`
✅ `sfdx force:mdapi:deploy -d converted/ -l RunLocalTests -w 0`
✅ `sfdx force:mdapi:deploy -d converted/ -l RunLocalTests`
✅ `sfdx force:mdapi:deploy -d converted/ -l RunLocalTests -w 100`
✅ `sfdx force:mdapi:deploy:report -i 0Af9A00000Z7DxUSAV -w 0`
✅ `sfdx force:mdapi:deploy:report -i 0Af9A00000Z7DxUSAV -w 0 --json`
✅ `sfdx force:mdapi:deploy:report -i 0Af9A00000Z7DxUSAV`
✅ `sfdx force:mdapi:deploy:report -i 0Af9A00000Z7DxUSAV -w -1`
✅ `sfdx force:mdapi:deploy:report -i 0Af9A00000Z7ETuSAN --json`
✅ `sfdx force:mdapi:deploy:report -i 0Af9A00000Z7FfXSAV -w 100`

✅ `sfdx force:mdapi:retrieve -r retrieved -k package.xml`
✅ `sfdx force:mdapi:retrieve -r retrieved -k package.xml -w -1`
✅ `sfdx force:mdapi:retrieve -r retrieved -k package.xml -w 0`
✅ `sfdx force:mdapi:retrieve:report -r retrieved -w 100 --json`
✅ `sfdx force:mdapi:retrieve:report -r retrieved -w 0`
✅ `sfdx force:mdapi:retrieve:report -r retrieved -w -1`
✅ `sfdx force:mdapi:retrieve:report`

**Pull request merged**